qa/workunits/rbd/cli_migration.sh

```diff
@@ -27,7 +27,10 @@ expect_false() {
 create_base_image() {
     local image=$1
 
-    rbd create --size 1G ${image}
+    # size is not a multiple of object size to trigger an edge case in
+    # list-snaps
```

Comment:
Is this the edge case that you refer to here https://tracker.ceph.com/issues/67845#note-1 ? 
"This is because deep_copy::ObjectCopyRequest submits list-snaps for 0~m_dst_image_ctx->layout.object_size range." I'm still unable to understand the cause of this edge case that you mention in the tracker comment. Can you dumb it down further?

Reply:
> Is this the edge case that you refer to here https://tracker.ceph.com/issues/67845#note-1 ? "This is because deep_copy::ObjectCopyRequest submits list-snaps for 0~m_dst_image_ctx->layout.object_size range."

Yes.  The edge case is an out of bounds read getting submitted to a stream, `ObjectCopyRequest` triggers it.

> I'm still unable to understand the cause of this edge case that you mention in the tracker comment. Can you dumb it down further?

Consider an image whose size is not a multiple of object size, e.g. 21M with standard 4M objects.  `ObjectCopyRequest` always submits list-snaps for entire objects, so when it gets to the last object, it will map 0..4M object extent to 20M..24M image extent and submit list-snaps for that.  If the image is being live imported from a raw format, `RawFormat::list_snaps()` will call `RawSnapshot::list_snap()` on the same extent and (with the current code -- this is getting changed in https://github.com/ceph/ceph/pull/44640) `SPARSE_EXTENT_STATE_DATA` will be returned for 20M..24M range.  `ObjectCopyRequest` will interpret that as an instruction to submit a read for 20M..24M range, generating an out-of-bounds read.

Does this help?

Reply:
Thanks for the explanation.  I understand it better now.